### PR TITLE
ConnectionModal uses Cond for rendering

### PR DESCRIPTION
**Pull request by masenf**
If we let the ConnectionModal use it's `is_shown` prop, then this adds 3ms to every re-render, which is about 20% of an average render (of a tiny page)

### CODE REVIEW

File: reflex/components/overlay/banner.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def create(cls, comp: Optional[Component] = None) -> Component:
         return super().create(has_connection_error, comp, Fragment.create())  # type: ignore
 
 
-class ConnectionModal(Modal):
+class ConnectionModal(Cond):
```
Comment (picklelo) [inline reflex/components/overlay/banner.py:66]: Should we just make this subclass Component and use a Cond inside the render?

File: reflex/components/overlay/banner.py [hunk 2]
```diff
@@ -79,7 +79,11 @@ def create(cls, comp: Optional[Component] = None) -> Component:
...
-            header="Connection Error",
-            body=comp,
-            is_open=has_connection_error,
+            has_connection_error,
+            Modal.create(
+                header="Connection Error",
+                body=comp,
+                is_open=has_connection_error,
+            ),
+            Fragment.create(),
```
Comment (picklelo) [inline reflex/components/overlay/banner.py:88]: Cond doesn't need a second argument

Comment (picklelo) [reply reflex/components/overlay/banner.py:88]: Though I noticed the type annotation for `Cond.create` is wrong as we don't set a default value of None - we should fix that.

Comment (masenf) [reply reflex/components/overlay/banner.py:88]: > we should fix that.

on it

File: reflex/components/layout/cond.py [hunk 3]
```diff
@@ -24,7 +24,10 @@ class Cond(Component):
 
     @classmethod
     def create(
-        cls, cond: Var, comp1: Component, comp2: Optional[Component]
+        cls,
+        cond: Var,
+        comp1: Component,
+        comp2: Optional[Component] = None,
```
Comment (picklelo) [inline reflex/components/layout/cond.py:30]: nit: we can do `Component | None` now

**Review by picklelo — APPROVED** [review]

**Pull request merged**